### PR TITLE
Fix Connect Wallet Functionality

### PR DESCRIPTION
**Pull request by math-marcellino**
1. Add 'Disconnect' button on navigation in Mobile Layout
2. Use 'disconnect' function from useAccount hook (wagmi) to disconnect user session on their wallet when disconnecting from mobile wallet (WalletConnect)
3. Add specific WC Connector for each network we support
4. Prevent user to connect with WalletConnect if the chosen network on the site is not the same as the one they choose on their wallet, but an error message will be thrown with toast instead.

### CODE REVIEW

**Comment by pyk** [comment]
@math-marcellino it seems that u are not using prettier to format the code.

U can use the following command to fix it:

```
npm run prettier-write
```

File: components/v1/Buttons/ConnectWalletDesktop.tsx [hunk 1]
```diff
@@ -82,6 +82,19 @@ const ButtonConnectWalletDesktop: FunctionComponent<ButtonConnectWalletDesktopPr
         setConnectorName(c.name);
 
         const result = await connect(c);
+
+        // Prevent connecting with WalletConnect if network is not right
+        if(c instanceof WalletConnectConnector){
+            if(result?.data?.chain?.unsupported || result?.data?.chain?.id !== chain.id){
```
Comment (pyk) [inline components/v1/Buttons/ConnectWalletDesktop.tsx]: @math-marcellino btw If we use this `result?.data?.chain?.id`, we can just use one `WCConnector` like before tho

Comment (math-marcellino) [reply components/v1/Buttons/ConnectWalletDesktop.tsx]: oh okay sir, I think I forgot that the one who automatically disconnect when the network is not supported & chainID not defined in WCConnector is useDApp not Wagmi xD, let me fix this again, Thank you!

**Review by pyk — APPROVED** [review]

**Comment by pyk** [comment]
Thanks @math-marcellino! Good work

**Pull request merged**